tests/snuba/api/endpoints/test_organization_events_trace.py

```diff
@@ -322,14 +326,121 @@ def test_third_generation_no_children(self):
 
         assert root_event_id in events
         event = events[root_event_id]
-        assert event["is_root"]
+        assert event["generation"] == 0
         assert event["parent_event_id"] is None
         assert event["parent_span_id"] is None
 
         assert current_event in events
         event = events[current_event]
-        assert not event["is_root"]
+        assert event["generation"] is None
         # Parent is unknown in this case
         assert event["parent_event_id"] is None
         # But we still know the parent_span
         assert event["parent_span_id"] == self.gen2_span_ids[0]
+
+
+class OrganizationEventsTrendsEndpointTest(OrganizationEventsTrendsEndpointBase):
+    url_name = "sentry-api-0-organization-events-trace"
+
+    def assert_trace_data(self, root):
+        """ see the setUp docstring for an idea of what the response structure looks like """
+        assert root["event_id"] == self.root_event.event_id
+        assert root["parent_event_id"] is None
+        assert root["parent_span_id"] is None
+        assert root["generation"] == 0
+        assert len(root["children"]) == 3
+
+        for i, gen1 in enumerate(root["children"]):
+            assert gen1["event_id"] == self.gen1_events[i].event_id
+            assert gen1["parent_event_id"] == self.root_event.event_id
+            assert gen1["parent_span_id"] == self.root_span_ids[i]
+            assert gen1["generation"] == 1
+            assert len(gen1["children"]) == 1
+
+            gen2 = gen1["children"][0]
+            assert gen2["event_id"] == self.gen2_events[i].event_id
+            assert gen2["parent_event_id"] == self.gen1_events[i].event_id
+            assert gen2["parent_span_id"] == self.gen1_span_ids[i]
+            assert gen2["generation"] == 2
+
+            # Only the first gen2 descendent has a child
+            if i == 0:
+                gen3 = gen2["children"][0]
+                assert gen3["event_id"] == self.gen3_event.event_id
+                assert gen3["parent_event_id"] == self.gen2_events[i].event_id
+                assert gen3["parent_span_id"] == self.gen2_span_ids[i]
+                assert gen3["generation"] == 3
+
+    def test_no_projects(self):
+        user = self.create_user()
+        org = self.create_organization(owner=user)
+        self.login_as(user=user)
+
+        url = reverse(
+            "sentry-api-0-organization-events-trace-light",
```

Comment:
Should this be `sentry-api-0-organization-events-trace`?

Reply:
guh, this should be `self.url_name` good catch